src/pudl/analysis/timeseries_cleaning.py

```diff
@@ -530,772 +726,932 @@ def impute_latc_tubal(  # noqa: C901
-        * code: https://github.com/truggles/EIA_Cleaned_Hourly_Electricity_Demand_Code
-        """
-        # Step 1
-        self.flag_negative_or_zero()
-        self.flag_identical_run(length=3)
-        self.flag_global_outlier(medians=9)
-        self.flag_global_outlier_neighbor(neighbors=1)
-        # Step 2
-        # NOTE: In original code, statistics used for the flags below are precomputed
-        # here, rather than computed for each flag with nulls added by previous flags.
-        window = 48
-        long_window = 480
-        iqr_window = 240
-        shifts = range(-240, 241, 24)
-        self.flag_local_outlier(
-            window=window,
-            shifts=shifts,
-            long_window=long_window,
-            iqr_window=iqr_window,
-            multiplier=(3.5, 2.5),
-        )
-        self.flag_double_delta(iqr_window=iqr_window, multiplier=2)
-        self.flag_single_delta(
-            window=window,
-            shifts=shifts,
-            long_window=long_window,
-            iqr_window=iqr_window,
-            multiplier=5,
-            rel_multiplier=15,
+    Examples:
+        >>> x = np.column_stack([[1, 2, np.nan, 4, 5, 6, 7, np.nan, np.nan]])
+        >>> simulate_nulls(x).ravel()
+        array([ True, False, False, False, True, True, False, False, False])
+        >>> simulate_nulls(x, lengths=[4], padding=0).ravel()
+        array([False, False, False, True, True, True, True, False, False])
+    """
+    new_nulls = np.zeros(x.shape, dtype=bool)
+    for col in range(x.shape[1]):
+        is_null = np.isnan(x[:, col])
+        if lengths is None:
+            run_values, run_lengths = encode_run_length(is_null)
+            run_lengths = run_lengths[run_values]
+        else:
+            run_lengths = lengths
+        is_new_null = insert_run_length(
+            new_nulls[:, col],
+            values=np.ones(len(run_lengths), dtype=bool),
+            lengths=run_lengths,
+            mask=None if overlap else ~is_null,
+            padding=0 if overlap else padding,
+            intersect=intersect,
         )
-        self.flag_anomalous_region(window=window + 1, threshold=0.15)
-
-    def summarize_flags(self) -> pd.DataFrame:
-        """Summarize flagged values by flag, count and median."""
-        stats = {}
-        for col in range(self.xi.shape[1]):
-            stats[self.columns[col]] = (
-                pd.Series(self.xi[:, col])
-                .groupby(self.flags[:, col])
-                .agg(["count", "median"])
-            )
-        df = pd.concat(stats, names=["column", "flag"]).reset_index()
-        # Sort flags by flagged order
-        ordered = df["flag"].astype(pd.CategoricalDtype(set(self.flagged)))
-        return df.assign(flag=ordered).sort_values(["column", "flag"])
+        if overlap:
+            is_new_null &= ~is_null
+        new_nulls[:, col] = is_new_null
+    return new_nulls
 
-    def plot_flags(self, name: Any = 0) -> None:
```

Comment:
I removed this plot flags method because it seems likely to be obsolete by the analysis notebooks we're going to create, but can add a version back in if this is useful